# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: intl/hyphenation/glue/nsHyphenationManager.h
# Commit: 4b2f994f3fe2
# Full Hash: 4b2f994f3fe2cad3895cc4d5044a3466a2d5eda4
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2020-10-12 15:27:19
# Regressor Bug: 1668253
# File Overlap Count: 1
# Description:
#   Bug 1668253 - Convert hyphenation from ipc::SharedMemoryBasic to base::SharedMemory, and freeze blocks before sharing to content processes. r=heycam
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D91945
# ==============================================================================

diff -r bc5b6dcb7057 -r 4b2f994f3fe2 intl/hyphenation/glue/nsHyphenationManager.h
--- a/intl/hyphenation/glue/nsHyphenationManager.h	Mon Oct 12 10:23:27 2020 +0000
+++ b/intl/hyphenation/glue/nsHyphenationManager.h	Mon Oct 12 10:24:06 2020 +0000
@@ -6,12 +6,12 @@
 #ifndef nsHyphenationManager_h__
 #define nsHyphenationManager_h__
 
-#include "nsInterfaceHashtable.h"
-#include "nsRefPtrHashtable.h"
+#include "base/shared_memory.h"
+#include "mozilla/Omnijar.h"
 #include "nsHashKeys.h"
+#include "nsInterfaceHashtable.h"
 #include "nsIObserver.h"
-#include "mozilla/Omnijar.h"
-#include "mozilla/ipc/SharedMemoryBasic.h"
+#include "nsRefPtrHashtable.h"
 
 class nsHyphenator;
 class nsAtom;
@@ -26,9 +26,9 @@
 
   already_AddRefed<nsHyphenator> GetHyphenator(nsAtom* aLocale);
 
-  void ShareHyphDictToProcess(
-      nsIURI* aURI, base::ProcessId aPid,
-      mozilla::ipc::SharedMemoryBasic::Handle* aOutHandle, uint32_t* aOutSize);
+  void ShareHyphDictToProcess(nsIURI* aURI, base::ProcessId aPid,
+                              base::SharedMemoryHandle* aOutHandle,
+                              uint32_t* aOutSize);
 
   static nsHyphenationManager* Instance();
 